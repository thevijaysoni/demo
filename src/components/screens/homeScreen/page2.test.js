import { render } from '@testing-library/react-native';
import { configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import React from 'react';
import { } from 'react-native';
import renderer from 'react-test-renderer';
import PageSecond from './page1';

jest.mock('../../common/RegularText', () => props => (
    <MockRegularText {...props} testID={"regular-text"} />
))
configure({ adapter: new Adapter() });

describe("PageSecond", () => {

    let props = {
        angerValue: 2,
        setAngerValue: jest.fn(),
    }

    beforeEach(() => {
    });

    it('should render PageSecond screen correctly', () => {
        const tree = renderer.create(<PageSecond {...props} />).toJSON();
        expect(tree).toMatchSnapshot();
    });

    it('should get 2 if angerValue is 2', () => {
        const { getByTestId } = render(<PageSecond {...props} />)
        expect(getByTestId('regular-text').props.children).toEqual(2)
    });

});