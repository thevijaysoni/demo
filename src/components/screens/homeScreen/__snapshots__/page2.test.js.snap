// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PageSecond should render PageSecond screen correctly 1`] = `
<View
  style={
    Object {
      "flex": 1,
      "justifyContent": "space-evenly",
      "marginHorizontal": 9,
      "marginTop": 18,
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "borderColor": "#ffffff",
        "borderRadius": 750,
        "borderStyle": "dashed",
        "borderWidth": 2,
        "height": 696,
        "justifyContent": "center",
        "marginBottom": 36,
        "width": 696,
      }
    }
  >
    <View
      indeterminateAnimationDuration={1000}
      style={
        Array [
          Object {
            "backgroundColor": "transparent",
            "overflow": "hidden",
          },
          Object {
            "alignItems": "center",
            "justifyContent": "center",
          },
        ]
      }
      testID="PageFirstCircleProgress"
    >
      <RNSVGSvgView
        bbHeight={522.9230769230769}
        bbWidth={522.9230769230769}
        collapsable={false}
        focusable={false}
        height={522.9230769230769}
        style={
          Array [
            Object {
              "backgroundColor": "transparent",
              "borderWidth": 0,
            },
            Object {},
            Object {
              "flex": 0,
              "height": 522,
              "width": 522,
            },
          ]
        }
        width={522.9230769230769}
      >
        <RNSVGGroup>
          <RNSVGPath
            d="M261.46153846153845 12
          A249.46153846153845 249.46153846153845 0 0 1 498.7135601803987 184.3736836340035"
            propList={
              Array [
                "stroke",
                "strokeWidth",
                "strokeLinecap",
              ]
            }
            stroke={4294967295}
            strokeLinecap={0}
            strokeWidth={24}
          />
        </RNSVGGroup>
      </RNSVGSvgView>
      <View
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "#386E7E",
            "borderRadius": 750,
            "height": 501.55555555555554,
            "justifyContent": "center",
            "position": "absolute",
            "width": 501.55555555555554,
          }
        }
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "#36DEE8",
              "borderRadius": 750,
              "height": 321,
              "justifyContent": "center",
              "width": 321,
            }
          }
        >
          <View
            style={
              Array [
                Object {
                  "justifyContent": "center",
                },
                undefined,
              ]
            }
          >
            <Text
              numberOfLines={100}
              style={
                Array [
                  Object {
                    "color": "#ffffff",
                    "fontSize": 100,
                  },
                  Object {
                    "fontFamily": "Poppins-SemiBold",
                  },
                ]
              }
              testID="regular-text"
            >
              2
            </Text>
          </View>
        </View>
      </View>
    </View>
  </View>
  <View
    style={
      Object {
        "alignItems": "center",
      }
    }
  >
    <View
      animationType="timing"
      onLayout={[Function]}
      onValueChange={[Function]}
      step={1}
      style={
        Array [
          Object {
            "height": 40,
            "justifyContent": "center",
          },
          Object {
            "width": "90%",
          },
        ]
      }
      thumbTouchSize={
        Object {
          "height": 40,
          "width": 40,
        }
      }
      value={2}
    >
      <View
        onLayout={[Function]}
        renderToHardwareTextureAndroid={true}
        style={
          Array [
            Object {
              "backgroundColor": "#ffffff",
            },
            Object {
              "borderRadius": 2,
              "height": 4,
            },
            Object {
              "borderRadius": 135,
              "height": 39.42857142857143,
            },
          ]
        }
      />
      <View
        collapsable={false}
        renderToHardwareTextureAndroid={true}
        style={
          Object {
            "backgroundColor": "#36DEE8",
            "borderRadius": 135,
            "height": 39.42857142857143,
            "opacity": 0,
            "position": "absolute",
            "width": 0,
          }
        }
      />
      <View
        collapsable={false}
        onLayout={[Function]}
        renderToHardwareTextureAndroid={true}
        style={
          Object {
            "backgroundColor": "#36DEE8",
            "borderColor": "#ffffff",
            "borderRadius": 135,
            "borderWidth": 6.571428571428571,
            "elevation": 5,
            "height": 49.285714285714285,
            "opacity": 0,
            "position": "absolute",
            "shadowColor": undefined,
            "shadowOffset": Object {
              "height": 1,
              "width": 1,
            },
            "shadowOpacity": 0.2,
            "shadowRadius": 5,
            "transform": Array [
              Object {
                "translateX": 0,
              },
              Object {
                "translateY": 0,
              },
            ],
            "width": 49.285714285714285,
          }
        }
      />
      <View
        onMoveShouldSetResponder={[Function]}
        onMoveShouldSetResponderCapture={[Function]}
        onResponderEnd={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderReject={[Function]}
        onResponderRelease={[Function]}
        onResponderStart={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        onStartShouldSetResponderCapture={[Function]}
        renderToHardwareTextureAndroid={true}
        style={
          Array [
            Object {
              "backgroundColor": "transparent",
              "bottom": 0,
              "left": 0,
              "position": "absolute",
              "right": 0,
              "top": 0,
            },
            Object {},
          ]
        }
      />
    </View>
  </View>
</View>
`;
